apps/zotonic_core/src/db/z_db_pgsql.erl

```diff
@@ -259,9 +260,11 @@ is_tracing() ->
 %% @doc This function MUST NOT be used, but currently is required by the
 %% install / upgrade routines. Can only be called from inside a
 %% z_db:transaction/2.
-get_raw_connection(#context{dbc=Worker}) when Worker =/= undefined ->
-    gen_server:call(Worker, get_raw_connection).
+get_raw_connection(Context) ->
+    get_raw_connection(Context, undefined).
 
+get_raw_connection(#context{dbc=Worker}, Timeout) when Worker =/= undefined ->
+    gen_server:call(Worker, {get_raw_connection, Timeout}).
```

Comment:
If you increase the timeout to get a connection, then this `gen_server:call` should also get timeout that is just a bit more.  Otherwise this call will timeout (default 5000 msec) before the timeout of the raw-connection fetch.